repo2docker/contentproviders/zenodo.py

```diff
@@ -0,0 +1,99 @@
+import os
+import json
+import shutil
+
+from os import makedirs
+from os import path
+from urllib.request import urlopen, Request
+from zipfile import ZipFile, is_zipfile
+
+from .base import ContentProvider
+from ..utils import copytree
+
+
+class Zenodo(ContentProvider):
+    """Provide contents of a Zenodo deposit."""
+
+    def detect(self, doi, ref=None, extra_args=None):
```

Comment:
I like this!

How can we support zenodo installations that aren't at zenodo.org?

Reply:
Good question. We'd have to maintain a list of DOI prefixes that we know map to a Zenodo instance and its hostname.

When we start doing this I'd also swap what `detect` returns to include the hostname instead of constructing the URL in `fetch`. I added a comment with this thinking in it.

I'd punt actually implementing it unless we know of an instance (I don't know any :( ).